### ISSUE TITLE
Feature request: Support for HA's new third party backup location integration

### ISSUE THREAD

**Issue by RubenKelevra** [sabeechen/hassio-google-drive-backup#1123]
I love your custom_component and have been using it for years now. I would really enjoy if you could create an integration for the new backup method provided by Home Assistant 2025.1, so we can sync our backups in the future to Google Drive directly within Home Assistant:

https://rc.home-assistant.io/blog/2024/12/23/release-20251/#backup-locations-are-extendable-by-design-

**Comment by Mariusthvdb** [comment]
there is something odd going on:

<img width="1328" alt="Scherm­afbeelding 2024-12-30 om 11 14 23" src="https://github.com/user-attachments/assets/64cc5082-0097-48f6-b686-ff43d7a5d39b" />

the new Automatic HA backups appear in the Google drive, and in the Add-on interface, the new automatic network storage backups show up:

<img width="860" alt="Scherm­afbeelding 2024-12-30 om 10 04 13" src="https://github.com/user-attachments/assets/9ca3cd52-3e0a-49d0-aeda-f6d90df64dc1" />

somehow the add-on confuses the HA system.

**Comment by Mariusthvdb** [comment]
sorry if I misunderstand your FR, but, in the current state of HA/Add-on, if you use the default location for both of the integrations, this is exactly what is happening.

HA stores the backup in the default location, as does the add-on.... next, the add-on uploads it to GD. be it the automatic backup, or a backup by the add-on

take care, this also happens when deleting backups.....

for now, I made a separate/dedicated network share on my system for the GD add-on, hoping this will keep the 2 apart.

if this is not at al what you need/request, please except my apologies for polluting your FR, and Ill delete the comments

**Comment by RubenKelevra** [comment]
@Mariusthvdb yes you misunderstood my FR. No worries, everything is fine.

**Comment by dougmaitelli** [comment]
I assume the FR is to change this addon to be a provider for GD to be used in the new automatic backups instead. There is probably no reason to have a separate automatic backup logic in an add-on anymore.

**Comment by RubenKelevra** [comment]
> I assume the FR is to change this addon to be a provider for GD to be used in the new automatic backups instead. There is probably no reason to have a separate automatic backup logic in an add-on anymore.

Yeah, I thought that's obvious?

**Comment by dougmaitelli** [comment]
> > I assume the FR is to change this addon to be a provider for GD to be used in the new automatic backups instead. There is probably no reason to have a separate automatic backup logic in an add-on anymore.
> 
> Yeah, I thought that's obvious? 

My comment was for @Mariusthvdb since he did misunderstood the original request.

**Comment by pantherale0** [comment]
I've built a backup provider for Google Drive here, its currently built to use Google Drives app data storage space so the only way to access the backups is via a Home Assistant instance.

https://github.com/pantherale0/ha-googledrive-backup

**Comment by dougmaitelli** [comment]
> I've built a backup provider for Google Drive here, its currently built to use Google Drives app data storage space so the only way to access the backups is via a Home Assistant instance.
> 
> https://github.com/pantherale0/ha-googledrive-backup

So, this means it is not possible to download the compressed zip backup from GD in order to restore an instance, you need to setup the integration first in order to restore, correct?

**Comment by pantherale0** [comment]
> So, this means it is not possible to download the compressed zip backup from GD in order to restore an instance, you need to setup the integration first in order to restore, correct?

Correct.

This is just how I've implemented it based on the scopes recommendations documented by Google. Plus the backups aren't actually named anything pretty anyway (simply their backup ID which is a 8 character hex string). There is also more metadata that is stored separately in this new backups mechanism in HA. I did try Google Drive's own metadata fields, but they don't support the complex types needed for the backups in HA.

Almost like you need a "database" of sorts alongside the backup files themselves. Want to see how this backup feature progresses though first before changing anything further, there is a lot of upset in the community by the forced encryption.

**Comment by rononline** [comment]
> I've built a backup provider for Google Drive here, its currently built to use Google Drives app data storage space so the only way to access the backups is via a Home Assistant instance.
> 
> https://github.com/pantherale0/ha-googledrive-backup

Not working for me, when I try to add the integration:
```
Error occurred loading flow for integration google_drive: No module named 'aiogoogle'
```

**Comment by pantherale0** [comment]
Ohhhh gutted, left some old refs in there then. Will fix now. 2 secs

**Comment by rononline** [comment]
That was quick! :-) And it worked now!

**Comment by dougmaitelli** [comment]
> > So, this means it is not possible to download the compressed zip backup from GD in order to restore an instance, you need to setup the integration first in order to restore, correct?
> 
> Correct.
> 
> This is just how I've implemented it based on the scopes recommendations documented by Google. Plus the backups aren't actually named anything pretty anyway (simply their backup ID which is a 8 character hex string). There is also more metadata that is stored separately in this new backups mechanism in HA. I did try Google Drive's own metadata fields, but they don't support the complex types needed for the backups in HA.
> 
> Almost like you need a "database" of sorts alongside the backup files themselves. Want to see how this backup feature progresses though first before changing anything further, there is a lot of upset in the community by the forced encryption.

I did not know they were forcing encryption.
I love HA, but they have some weird decisions sometimes, like the resistance on adopting support to external authentication layers like OpenID and other things.

Definitely makes sense to wait a while then and see how this evolves.

**Comment by pantherale0** [comment]
Someone is working on this to be added into core, perhaps for release 2024.2.0

**Comment by mgriffin13** [comment]
> Someone is working on this to be added into core, perhaps for release 2024.2.0
Do you have an issue that I could watch? I tried searching the open issues but I wasn't able to find anything

**Comment by pantherale0** [comment]
Nope, it was mentioned in the beta channel of the discord server.

**Comment by ShonP40** [comment]
> I've built a backup provider for Google Drive here, its currently built to use Google Drives app data storage space so the only way to access the backups is via a Home Assistant instance.
> 
> https://github.com/pantherale0/ha-googledrive-backup

Kinda impossible to install it since you archived the repo…

**Comment by pdecat** [comment]
The HA core PR is at https://github.com/home-assistant/core/pull/134576

**Comment by Mariusthvdb** [comment]
we should probably keep the FR  in this Repo about the add-on and not discuss the aspects of the new PR's for HA here.
Great that is being developed btw. but no subject in here, if only to keep things tidy.

the add-on has so much to offer that probably will never get accepted in HA core, so FR ing this add-on is very valuable as it stands.

**Comment by gfn256** [comment]
Following this thread with interest, however it appears that sabeechen has gone AWOL/lost interest in this project.
Possibly because he is aware of the newly (2025.1) implemented core backup feature set?

**Comment by RubenKelevra** [comment]
> Following this thread with interest, however it appears that sabeechen has gone AWOL/lost interest in this project.
> Possibly because he is aware of the newly (2025.1) implemented core backup feature set?

To be fair it's holiday time. So I wouldn't interpret too much into that. :)

**Comment by gfn256** [comment]
> To be fair it's holiday time. So I wouldn't interpret too much into that. :)

Search his activity on this thread - has not been seen for months.

**Comment by sergeykad** [comment]
He made 14 contributions already in 2025 according to Github :)
https://github.com/sabeechen?tab=overview&from=2025-01-01&to=2025-01-05

**Comment by gfn256** [comment]
> He made 14 contributions already in 2025 according to Github :)

I said on this thread - [HA GDB](https://github.com/sabeechen/hassio-google-drive-backup)

**Comment by gfn256** [comment]
The fact that @sabeechen has not responded to this thread & others of GDB; I think shows that this repo as we know it is being deprecated. (I'm not even sure if he is even following it).

This is a very sad day. I've come to rely on it over the years - and in my opinion was the best ever add-on. It just worked & got the job done. sabeechen got through the hurdles of accessing Google's ecosystem with ease, worked through the kinks when Google Co. made changes along the way, eased us through HA updates when necessary, and in general provided a top-notch helping hand, with a simple, robust & invaluable service. Kudos to him.

I imagine something similar will eventually (when?) be incorporated into the new core backup system - but I have my doubts if it will at all match this top-notch add-on as we know it.

I really would appreciate renewed interest in this repo.

**Comment by sabeechen** [comment]
I'm still here, I just don't respond very often and focus on maintenance rather than new features.  I'll continue to keep this addon running for as long as HA doesn't deprecate the API features it needs to work.

The new "Backup API" they're making looks great to me but as others have pointed out this addon has a whole complex feature set (Generational backup, local history management, addon stopping, etc) on the side that I don't think they'd want to support.  

The PR to [add Google Drive as an integartion](https://github.com/home-assistant/core/pull/134576) looks spiffy to me, though it still requires setting up developer credentials with google which is beyond what most users are willing to do.  If they were able to develop around that then I imagine it would handle the needs of like 90% of the users of this addon.

So ... HA is clearly going its own way with respect to the new automatic backup feature+Google Drive.  I'll keep this addon around, as-is, until HA's API changes make me unable to do so, for those who have come to depend on its esoteric features.

**Comment by gfn256** [comment]
Thanks @sabeechen for that lease-of-life! I have already elaborated above on the virtues of your add-on & I'll just add the following.
In my opinion there are few if any add-ons that are simply so perfect that they warrant to be imported as a core integrated feature; if GDB is not one of those then there are none that qualify! I really expected over the years, for this to be simply replaced as-is to an integration within HA.



> for those who have come to depend on its esoteric features

"esoteric" as defined in the Oxford dictionary: _**"intended for or likely to be understood by only a small number of people with a specialized knowledge or interest".**_
I can't find any stats/analytics on the number of users who use this add-on & its' awesome features, but based on the number of mentions in forums & other places on the web - I would not describe this add-on as "esoteric" but rather "invaluable".

Thanks again!
